homeassistant/components/hausbus/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Config flow for Haus-Bus integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from pyhausbus.BusDataMessage import BusDataMessage
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.data.ModuleId import ModuleId
+from pyhausbus.HomeServer import HomeServer
+from pyhausbus.IBusDataListener import IBusDataListener
+from pyhausbus.ObjectId import ObjectId
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_SCHEMA = vol.Schema({})
+
+
+class ConfigFlow(IBusDataListener, config_entries.ConfigFlow, domain=DOMAIN):  # type: ignore[misc]
+    """Handle a config flow for hausbus."""
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._found_device = False
+        self._search_task: asyncio.Task | None = None
+        self.home_server = HomeServer()
+        self.home_server.addBusEventListener(self)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # start searching for devices
+            return await self.async_step_wait_for_device()
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_SCHEMA, errors=errors
+        )
+
+    async def async_step_wait_for_device(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Wait for a hausbus device to be found."""
+        if not self._search_task:
+            self._search_task = self.hass.async_create_task(
+                self._async_wait_for_device()
+            )
+
+        if not self._search_task.done():
+            return self.async_show_progress(
+                step_id="wait_for_device",
+                progress_action="wait_for_device",
+                progress_task=self._search_task,
+            )
+
+        try:
+            await self._search_task
+        except TimeoutError:
+            return self.async_show_progress_done(next_step_id="search_timeout")
+        finally:
+            self._search_task = None
+
+        return self.async_show_progress_done(next_step_id="search_complete")
+
+    async def async_step_search_timeout(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Inform the user that no device has been found."""
+        if user_input is not None:
+            return await self.async_step_wait_for_device()
+
+        self._set_confirm_only()
```

Comment:
Why do we set this here?

Reply:
I copied this from an other integration, will check if this is needed at all.